content/posts/2020/12/libplanet-0.10/index.kor.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Libplanet 0.10 릴리스
+date: 2020-12-04
+authors: [suho.lee]
+---
+
+안녕하세요.  [Libplanet]의 열 번째 마이너 버전인 [0.10 버전][0.10.0]이
+릴리스되었습니다.
+
+Libplanet은 분산 P2P로 돌아가는 온라인 멀티플레이어 게임을 만들 때,
+그러한 게임들이 매번 구현해야 하는 P2P 통신이나 데이터 동기화 등의 문제를 푸는
+공용 라이브러리입니다.
+
+이번 버전부터 Libplanet은 상태를 관리하기 위해 [Merkle Patricia Trie(MPT)][MPT]를
+사용하고, 자산 관리를 위한 별도의 API가 추가되는 등, 여기서 소개하는 내용 이외에도
+인터페이스 내외로 많은 변경점이 있었습니다.
+
+이 글에서는 0.10 버전의 주요 변경 사항들에 대해서 다루겠습니다.
+
+[Libplanet]: https://libplanet.io/
+[0.10.0]: https://github.com/planetarium/libplanet/releases/tag/0.10.0
+[MPT]: https://eth.wiki/en/fundamentals/patricia-tree 
+
+[Block<T>.PreEvaluationHash]
+---------------------------
+
+그동안 [`Block<T>`] 는 해당 블록이 가지고 있는 상태에 대한 정보를 따로 들고 있지 않았습니다.
+따라서 상태에서 블록을 유도할 수는 있어도, 블록에서 상태의 정합성 등을 검증할 수 있는 방법은
+오직 액션을 직접 실행하는 방법 뿐이었습니다. 하지만 이제 [`Block<T>.Hash`] 는 해당 블록에 대한 정보 뿐만 아니라, 
+액션을 평가하고 나온 결과의 `Hash` 도 포함되어 유도됩니다. 이전과 같이 액션을 평가하지 않고 블록 정보만을 가진
+`Hash` 는 `Block<T>.PreEvaluationHash` 로 저장됩니다.
+
+[`Block<T>`]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html
+[Block<T>.PreEvaluationHash]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_PreEvaluationHash
+[`Block<T>.Hash`]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_Hash
+
+[Block<T>.TotalDifficulty]
+---------------------------
+
+그동안은 블록체인의 정본(正本, Canonical chain)을 고르는 데 [`Block<T>.Index`]가 사용되었습니다.
+하지만 이 방법은 리오그가 발생해도 블록 인덱스가 같은 경우가 자주 발생하여, 장기간 리오그가 일어나지 않다가
+한 번에 깊은 리오그가 발생하는 일이 잦았습니다. 따라서 저희는 `Block<T>.TotalDifficulty` 를 정본을 고르는 기준으로
+삼게 되었습니다.
+
+[Block<T>.TotalDifficulty]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_TotalDifficulty
+[`Block<T>.Index`]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_Index 
+
+[Merkle Patricia Trie (MPT)]
+----------------------------
+
+Merkle Patricia Trie (이하 MPT)는 Ethereum 등지에서 상태를 관리하는 데 사용되는 트리 구조입니다.
```

Comment:
```
➜  snack.planetarium.dev git:(release-0.10.1) ✗ grep -r "이더" .
./content/posts/2019/05/first-contribution.kor.md:Libplanet 주소는 [이더리움][5]의 주소형식을 차용하고 있습니다. 이더리움 주소는 [EIP-55][]에서 제안된 대소문자를 혼합한 형식의 체크섬을 사용하고 있는데, 이 주소 형식의 예제는 아래와 같습니다.
./content/posts/2019/05/first-contribution.kor.md:당시에 Libplanet 주소의 문자열 표현은 소문자만으로 구성된 십육진법이었습니다. 따라서 주소의 문자열 표현을 이더리움에서 사용하고 있는 EIP-55 체크섬 형식으로 표현되도록 구현해야 하는 이슈였습니다. 자세한 내용은 해당 [이슈][6]와 [PR][7]을 참고하실 수 있습니다.
./content/posts/2019/11/libplanet-0.7/index.kor.md:이번 버전에서는 개인키를 암호화하여 안전하게 보호할 수 있도록 키 저장소가 추가되었습니다. 키 저장소 내 각각의 키 파일은 [`ProtectedPrivateKey`][ProtectedPrivateKey] 클래스로 표현되며, 사용자가 입력한 암호(passphrase)로 개인키를 암호화 하여 저장할 수 있습니다. 또한, [`ProtectedPrivateKey.WriteJson()`][ProtectedPrivateKey.WriteJson] 메서드를 이용해 [이더리움][Ethereum]의 [Web3 Secret Storage Definition]에 따라 JSON 형식으로 저장할 수 있습니다. 추후 키 저장소 디렉터리를 통합적으로 관리하는 기능도 추가될 예정입니다.
```
```
➜  snack.planetarium.dev git:(release-0.10.1) ✗ grep -r "Ethereum" .
./content/posts/2019/05/first-contribution.eng.md:Libplanet’s address uses [Ethereum][5]'s address format, which uses a checksum in a combination of upper and lower case letters proposed in [EIP-55][]. For example:
./content/posts/2019/05/first-contribution.eng.md:When I first contributed to Libplanet, the project used hexadecimal consisting of only lowercase letters for its address format. Therefore, this issue had to be implemented so that the address string representation was expressed in the EIP-55 checksum format used by Ethereum. For more information, see the corresponding [issue][6] and [pull request][7].
./content/posts/2019/11/libplanet-0.7/index.eng.md:In this version, we have added a Key Store to help you encrypt and secure your private key. Each key file in the Key Store is represented by the [`ProtectedPrivateKey`] [ProtectedPrivateKey] class and can be saved by encrypting the private key with the passphrase you enter. Also, you can save it in JSON format based on [Web3 Secret Storage Definition] of [Ethereum] using the method [`ProtectedPrivateKey.WriteJson()`][ProtectedPrivateKey.WriteJson]. In the future, there will be additional features for integrated management of Key Store directory.
./content/posts/2019/11/libplanet-0.7/index.eng.md:[Ethereum]: https://en.wikipedia.org/wiki/Ethereum
./content/posts/2019/11/libplanet-0.7/index.kor.md:이번 버전에서는 개인키를 암호화하여 안전하게 보호할 수 있도록 키 저장소가 추가되었습니다. 키 저장소 내 각각의 키 파일은 [`ProtectedPrivateKey`][ProtectedPrivateKey] 클래스로 표현되며, 사용자가 입력한 암호(passphrase)로 개인키를 암호화 하여 저장할 수 있습니다. 또한, [`ProtectedPrivateKey.WriteJson()`][ProtectedPrivateKey.WriteJson] 메서드를 이용해 [이더리움][Ethereum]의 [Web3 Secret Storage Definition]에 따라 JSON 형식으로 저장할 수 있습니다. 추후 키 저장소 디렉터리를 통합적으로 관리하는 기능도 추가될 예정입니다.
./content/posts/2019/11/libplanet-0.7/index.kor.md:[Ethereum]: https://en.wikipedia.org/wiki/Ethereum
```

'이더리움' 이라고 그동안 표기해 왔던 것 같습니다. 한글 표기로 고치겠습니다.